### PR TITLE
Refactor/mario class and forward fill

### PR DESCRIPTION
**Pull request by icebound777**
This refactor aims to organize the codebase a bit by doing the following:

- Remove some deprecated or dead code, like Forward Fill or alpha item prices
- Consolidate simulate.py and relevant code from other modules into one MarioInventory model

### CODE REVIEW

File: models/MarioInventory.py [hunk 1]
```diff
@@ -1,30 +1,122 @@
...
+        """
+        Initializes Mario's starting inventory with values depending on
+        several parameters and flags.
+        This includes partners (for considering their overworld abilities during
+        world graph traversal), equipment, as well as additional pseudoitems
+        based on chosen settings.
+        Starting equipment irrelevant to world graph traversal (such as Lucky
+        Star, starting coins) is ignored.
+        """
+        assert(isinstance(starting_boots, int) and starting_boots in range(-1,3))
```
Comment (Pronyo-Chan) [inline models/MarioInventory.py:37]: Shouldn't the range be (-1,2)?

**Review by Pronyo-Chan — APPROVED** [review]
Great cleanup

File: models/MarioInventory.py (hunk 1 above)
Comment (icebound777) [reply models/MarioInventory.py:37]: python's range function includes the first, but excludes the second parameter. so range(-1,3) produces [-1,0,1,2]

Comment (Pronyo-Chan) [reply models/MarioInventory.py:37]: I always forget this weirdness, great then

**Pull request merged**